pkg/cmd/pulumi/new_test.go

```diff
@@ -962,3 +984,40 @@ func TestProjectExists(t *testing.T) {
 		})
 	}
 }
+
+//nolint:paralleltest // changes directory for process
+func TestGenerateOnlyProjectCheck(t *testing.T) {
+	// Regression test for https://github.com/pulumi/pulumi/issues/13527, make sure various combinations of
+	// project name and stack work when generateOnly is set (thus skipping backend checks).
+
+	cases := []struct {
+		name  string
+		stack string
+	}{
+		{name: "mismatched project name", stack: "organization/boom/stack"},
+		{name: "fully qualified stack name", stack: "organization/project/stack"},
+		{name: "org qualified stack name", stack: "organization/stack"},
+		{name: "unqualified stack name", stack: "stack"},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
```suggestion
	for _, tt := range cases {
		tt := tt
		t.Run(tt.name, func(t *testing.T) {
```

Reply:
Sure, not strictly needed because we're not parallel here but keeps consistency.